src/calliope/postprocess/orchestrator.py

```diff
@@ -0,0 +1,175 @@
+
+
+class PostprocessSettings(BaseModel):
+    """Schema for postprocessing function settings."""
+
+    model_config = ConfigDict(frozen=True, extra="forbid")
+
+    name: AttrStr
+    order: int
+    base_math: NonEmptyUniqueList
+    enabled: bool
+
+
+class PostprocessRegistry:
+    """Postprocessing registry class."""
+
+    def __init__(self) -> None:
+        """Initialise with an empty registry."""
+        self._entries: dict[str, tuple[PostprocessSettings, PostprocessFunction]] = {}
+        self._lock = threading.Lock()
+
+    def register_function(
+        self,
+        function: PostprocessFunction,
+        *,
+        name: str,
+        base_math: Iterable[str],
+        order: int,
+        enabled: bool,
+        overwrite: bool,
+    ) -> None:
+        """Add an entry to the postprocessing registry."""
+        with self._lock:
+            spec = PostprocessSettings(
+                name=name, order=order, base_math=listify(base_math), enabled=enabled
+            )
+            if spec.name in self._entries and not overwrite:
+                raise ValueError(f"Postprocessor '{spec.name}' is already registered.")
+            self._entries[spec.name] = (spec, function)
+
+    def get_applicable(
+        self, base_math: str
+    ) -> list[tuple[PostprocessSettings, PostprocessFunction]]:
+        """Find postprocessing function entries compatible with this base math file."""
+        with self._lock:
+            items = [
+                (spec, fn)
+                for spec, fn in self._entries.values()
+                if (base_math in spec.base_math) and (spec.enabled)
+            ]
+            items.sort(key=lambda t: (t[0].order, str(t[0].name)))
+        return items
+
+    def set_enabled(self, name: str, enabled: bool) -> None:
+        """Enable or disable a specific function."""
+        with self._lock:
+            spec, fn = self._entries[name]
+            self._entries[name] = (spec.model_copy(update={"enabled": enabled}), fn)
+
+
+REGISTRY: Final = PostprocessRegistry()
+
+
+def postprocessor(
+    *,
+    name: str | None = None,
+    base_math: Iterable[str],
+    order: int,
+    enabled: bool,
+    overwrite: bool = False,
+) -> Callable[[F], F]:
+    """Decorator used to add new functions to the postprocessing registry.
+
+    Args:
+        base_math (Iterable[str]): compatible base math files for this function.
+        name (str, optional): name to use for this post processing function. Defaults to the function name.
+        order (int): priority order.
+        enabled (bool): whether or not this function should be enabled.
+        overwrite (bool, optional): whether or not to allow overwriting functions. Defaults to False.
+    """
+
+    def decorator(func: F) -> F:
+        REGISTRY.register_function(
+            func,
+            base_math=base_math,
+            name=name or func.__name__,
+            order=order,
+            enabled=enabled,
```

Comment:
what's the value of `enabled`? In the core code at least there's no way of anyone accessing it...